Automattic/wp-calypso
# Plans 2023: Refactor interval type toggle on change handler

Related to https://github.com/Automattic/wp-calypso/pull/86127

[We refactored onChange handling for the new IntervalTypeDropdown component]( https://github.com/Automattic/wp-calypso/pull/86127 ). This PR is a follow-up, because we had planned to do the same for the IntervalTypeToggle component. See https://github.com/Automattic/wp-calypso/pull/86127#discussion_r1449253021

## Proposed Changes

* Simplify the plan type toggle and make it more of a purely presentational component by shifting any url generation concerns to consuming components
* Move any relative path or query string generation from either plan type selector to Plans Features Main
* Expose a callback that's invoked on plan type selection. Pass simple keys to callback and allow consuming components to decide what to do with the information
* The `isPlansInsideStepper` prop is unnecessary now for the plan term selector. It was used to conditionally render [a react-router-dom Link component instead of a traditional anchor tag](https://github.com/Automattic/wp-calypso/pull/87298/files#diff-9ab80a88f1adeb05704b085d9db89e1bf7f79f60f58acdb31329864cf5ed9ea9L6-L11 ) Routing logic is now [handled by consuming components](https://github.com/Automattic/wp-calypso/blob/trunk/client/landing/stepper/declarative-flow/internals/steps-repository/plans/plans-wrapper.tsx#L117-L125)

## Testing Instructions



* Check out branch
* Navigate to `/start/plans` and smoke test the plan term dropdown
* At this point, the easiest way to find a plan term toggle component is to disable the "onboarding/interval-dropdown" config flag. Feel free to do this by setting the value for "onboarding/interval-dropdown" in the `config/development.json` file to false
* Navigate to `/start/plans`
* Confirm that the toggle behaves as expected
* Do the same for the WooExpress plans page ( see PCYsg-RDT-p2 for instructions )

## Pre-merge Checklist
